### PR TITLE
DS-10153: Add Functionality to aws client for calling S3GetAccessPointPolicy

### PR DESCRIPTION
**Pull request by tolujimoh**
### Description

This PR adds functionality for calling S3GetAccessPointPolicy to support [clutch-private#2500](https://github.com/lyft/clutch-private/pull/2500)

### CODE REVIEW

File: backend/service/aws/aws.go [hunk 1]
```diff
@@ -178,6 +180,8 @@ type Client interface {
 	S3GetBucketPolicy(ctx context.Context, account, region, bucket, accountID string) (*s3.GetBucketPolicyOutput, error)
 	S3StreamingGet(ctx context.Context, account, region, bucket, key string) (io.ReadCloser, error)
 
+	S3GetAccessPointPolicy(ctx context.Context, account, region, accessPointName string, accountID string) (*s3control.GetAccessPointPolicyOutput, error)
+
```
Comment (mikecutalo) [inline backend/service/aws/aws.go:184]:
```suggestion
	S3GetAccessPointPolicy(ctx context.Context, account, region, accessPointName, accountID string) (*s3control.GetAccessPointPolicyOutput, error)
```

consecutive params of the same type only need a single type definition

File: backend/service/aws/aws.go [hunk 2]
```diff
@@ -263,6 +269,8 @@ func (r *regionalClient) S3() *s3.Client {
 	return r.s3.(*s3.Client)
 }
 
+func (r *regionalClient) S3Control() *s3control.Client { return r.s3control.(*s3control.Client) }
+
```
Comment (mikecutalo) [inline backend/service/aws/aws.go:273]: We already expose all clients via the `GetDirectClient()` on the aws client struct, we dont need to expose them individually.

**Review by mikecutalo — COMMENTED** [review]
Nice! I just did a first pass and left a few comments.

File: backend/service/aws/aws.go (hunk 1 above)
Comment (tolujimoh) [reply backend/service/aws/aws.go]: Good catch! Updated.

File: backend/service/aws/aws.go (hunk 2 above)
Comment (tolujimoh) [reply backend/service/aws/aws.go:272]: Ah! Thanks for pointing, will update.

**Review by mikecutalo — APPROVED** [review]
Approved!

**Pull request merged**